modules/data.atmosphere/R/download.CRUNCEP_Global.R

```diff
@@ -60,12 +65,27 @@ download.CRUNCEP <- function(outfolder, start_date, end_date, site_id, lat.in, l
     ntime <- ifelse(lubridate::leap_year(year), 366 * 4, 365 * 4)
 
     loc.file <- file.path(outfolder, paste("CRUNCEP", year, "nc", sep = "."))
+    results$file[i] <- loc.file
+    results$host[i] <- PEcAn.utils::fqdn()
+    results$startdate[i] <- paste0(year, "-01-01 00:00:00")
+    results$enddate[i] <- paste0(year, "-12-31 23:59:59")
+    results$mimetype[i] <- "application/x-netcdf"
+    results$formatname[i] <- "CF Meteorology"
+
+    if (file.exists(loc.file) && !isTRUE(overwrite)) {
+      logger.error("File already exists. Skipping to next year")
+      next
+    }
+
     PEcAn.utils::logger.info(paste("Downloading",loc.file))
     ## Create dimensions
     lat <- ncdf4::ncdim_def(name = "latitude", units = "degree_north", vals = lat.in, create_dimvar = TRUE)
     lon <- ncdf4::ncdim_def(name = "longitude", units = "degree_east", vals = lon.in, create_dimvar = TRUE)
-    time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:ntime) * 21600, 
-                      create_dimvar = TRUE, unlim = TRUE)
+
+    timestamps <- (1:ntime) * 6/24 - 3/24 # data are 6-hourly, with timestamp at center of interval
```

Comment:
We're keeping the timestamp. Or more precisely, we're generating a completely new time series here, then on reading each variable in the loop below we check that input times have the same values *after converting to our output units*.

Input timestamps of CRUNCEP nc files appear to always have units "days since 1700-01-01T00:00:00Z", so a file starting 2004-01-01 looks like `111033.1 111033.4 111033.6 111033.9 111034.1 ...`. Output timestamps are days since `start_date`, so the times always look like `0.125 0.375 0.625 0.875 1.125 ...`.